Kitodo/src/main/java/org/kitodo/services/data/ImportService.java

```diff
@@ -0,0 +1,197 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jdom2.JDOMException;
+import org.jdom2.input.SAXBuilder;
+import org.jdom2.output.DOMOutputter;
+import org.kitodo.api.externaldatamanagement.ExternalDataImportInterface;
+import org.kitodo.api.externaldatamanagement.SearchResult;
+import org.kitodo.config.ConfigCore;
+import org.kitodo.config.OPACConfig;
+import org.kitodo.config.enums.ParameterCore;
+import org.kitodo.serviceloader.KitodoServiceLoader;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+
+public class ImportService {
+
+    private static final Logger logger = LogManager.getLogger(ImportService.class);
+
+    private static ImportService instance = null;
+    private static ExternalDataImportInterface importModule;
+
+    /**
+     * Return singleton variable of type ImportService.
+     *
+     * @return unique instance of ImportService
+     */
+    public static ImportService getInstance() {
+        if (Objects.equals(instance, null)) {
+            synchronized (ImportService.class) {
+                if (Objects.equals(instance, null)) {
+                    instance = new ImportService();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Load ExternalDataImportInterface implementation with KitodoServiceLoader and perform given query string
+     * with loaded module.
+     *
+     * @param searchField field to query
+     * @param searchTerm  given search term
+     * @param catalogName catalog to search
+     * @return search result
+     */
+    public SearchResult performSearch(String searchField, String searchTerm, String catalogName) {
+        importModule = initializeImportModule();
+        try {
+            OPACConfig.getOPACConfiguration(catalogName);
+        } catch (IllegalArgumentException e) {
+            logger.error(e.getLocalizedMessage());
+            throw new IllegalArgumentException("Error: OPAC '" + catalogName + "' is not supported!");
+        }
+        return importModule.search(catalogName, searchField, searchTerm, 10);
+    }
+
+    private ExternalDataImportInterface initializeImportModule() {
+        KitodoServiceLoader<ExternalDataImportInterface> loader =
+                new KitodoServiceLoader<>(ExternalDataImportInterface.class);
+        return loader.loadModule();
+    }
+
+    /**
+     * Load search fields of catalog with given name 'opac' from OPAC configuration file and return them as a list
+     * of Strings.
+     *
+     * @param opac name of catalog whose search fields are loaded
+     * @return list containing search fields
+     * @throws IllegalArgumentException thrown when configuration for catalog with name 'opac' cannot be found in
+     *                                  opac configuration file
+     */
+    public List<String> getAvailableSearchFields(String opac) throws IllegalArgumentException {
```

Comment:
No need to define "throws IllegalArgumentException" in function. It extends RuntimeException.

`public List<String> getAvailableSearchFields(String opac) {`

Reply:
Done.